ambry-api/src/main/java/com.github.ambry/clustermap/DiskId.java

```diff
@@ -39,4 +42,9 @@
    * @return the raw capacity in bytes
    */
   long getRawCapacityInBytes();
+
+  /**
+   * @return a snapshot which includes information that the implementation considers relevant.
+   */
+  JSONObject getSnapshot();
```

Comment:
Why not add this to the Resource superinterface?  Does any subinterface not need this?
Even if not every class needs it, we can add an intermediate interface like SnapshottableResource.

Reply:
Makes sense - will look at this

Reply:
Did this in #1065